darshan-util/pydarshan/darshan/experimental/plots/plot_common_access_table.py

```diff
@@ -122,14 +121,14 @@ def __init__(self, df: Any, **kwargs):
         self.html = self.df.to_html(**kwargs)
 
 
-def plot_common_access_table(report: darshan.DarshanReport, mod: str, n_rows: int = 4) -> DarshanReportTable:
+def plot_common_access_table(record: dict, mod: str, n_rows: int = 4) -> DarshanReportTable:
     """
     Creates a table containing the most
     common access sizes and their counts.
 
     Parameters
     ----------
-    report: a ``darshan.DarshanReport``.
+    record: a dict.
```

Comment:
In cffi_backend.py, rec_dict is used for the accumulate_records function. maybe we could use a name like acc_rec_dict?

Reply:
Yeah, maybe? I was just thinking we kind of have a custom object that subclasses `dict` or something like that, and then it is clear to folks what the data structure should be since we define it ourselves somewhere. Probably a minor point that you could easily change later on though, so probably not a huge priority compared to dealing with all the testing/code changes.

```python
class acc_rec_dict(dict):
    # stores darshan counter and fcounter dataframes
    pass
```

Or maybe there's a better idea?

I also wonder if it might make sense to provide some way to i.e., automatically produce such an object from multiple logs in a convenience function like `produce_acc_rec_dict(list_of_logs)`. But that may also be something we could think about later on, if you and Shane haven't already.

Reply:
I'm on board with coming up with a more specific class for a "record" type, even if it's just a dict subclass. I don't know if the DarshanRecordCollection thing we added is somehow useful for this or if that's meant to be hidden from users -- probably a better question for @jakobluettgau .

I think we're on the same page, but I'll clarify just to double check:

- These routines should take a record-like object that has counters/fcounters data frames internally.
- This record structure could correspond to a single file (i.e., a single file record from the set of file records a module collects) or it could correspond to arbitrary accumulations of record sets (i.e., all records for a given module in a single log, all records for a given module across multiple logs).

> I also wonder if it might make sense to provide some way to i.e., automatically produce such an object from multiple logs in a convenience function like produce_acc_rec_dict(list_of_logs). But that may also be something we could think about later on, if you and Shane haven't already.

Maybe? At the very least, it might make sense to make the existing API (`accumulate_records()` in cffi backend) a little more flexible. It currently takes in a set of records and accumulates the entire set (with this set of records produced by the `to_df()` routine). But, to make this work across multiple logs we'd have to find a way to munge the record sets from each log into a single structure... Instead, it might make sense to provide an API that users can call repeatedly with new record sets, and then an additional routine to pull out the final result (derived metrics + summary record tuple).

Reply:
yeah, we're on the same page; for my other comment below I imagine you'll just tag on to the C accum infrastructure for now, but I felt compelled to whine :)